components/nodemanager-service/pgdb/project_update.go

```diff
@@ -0,0 +1,281 @@
+package pgdb
+
+import (
+	"context"
+
+	"encoding/json"
+
+	iam_v2 "github.com/chef/automate/api/interservice/authz/v2"
+	"github.com/chef/automate/components/nodemanager-service/api/nodes"
+	project_update_lib "github.com/chef/automate/lib/authz"
+	"github.com/chef/automate/lib/stringutils"
+	uuid "github.com/chef/automate/lib/uuid4"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+const selectNodesProjectData = `
+SELECT
+  n.id,
+  n.projects_data
+FROM nodes n
+`
+
+type NodeProjectData struct {
+	ID                   string `db:"id"`
+	Data                 string `db:"projects_data"`
+	ProjectDataKeyValues []*nodes.ProjectsData
+}
+
+// JobCancel - cancel the a currently running project update
+func (db *DB) JobCancel(ctx context.Context, projectID string) error {
+	log.Debugf("Node manager project update JobCancel %s", projectID)
+	if db.ProjectUpdate.ID == projectID {
+		db.ProjectUpdate.running = false
+	}
+	return nil
+}
+
+// UpdateProjectTags - start a project update
+func (db *DB) UpdateProjectTags(ctx context.Context,
+	projectRules map[string]*iam_v2.ProjectRules) ([]string, error) {
+	jobUUID, _ := uuid.NewV4()
+	log.Debugf("Running node manager UpdateProjectTags %v ID: %s", projectRules, jobUUID.String())
+
+	// setup for new project update run
+	db.ProjectUpdate.jobStatusError = nil
+	db.ProjectUpdate.running = true
+	db.ProjectUpdate.jobStatus = project_update_lib.JobStatus{
+		Completed:             false,
+		PercentageComplete:    0,
+		EstimatedEndTimeInSec: 0,
+	}
+	db.ProjectUpdate.ID = jobUUID.String()
+	go db.updateNodes(projectRules)
+
+	return []string{jobUUID.String()}, nil
+}
+
+// JobStatus - get the job status of the project update
+func (db *DB) JobStatus(ctx context.Context, projectID string) (project_update_lib.JobStatus, error) {
+	if db.ProjectUpdate.ID == projectID {
+		if db.ProjectUpdate.jobStatusError != nil {
+			return project_update_lib.JobStatus{}, db.ProjectUpdate.jobStatusError
+		}
+		return db.ProjectUpdate.jobStatus, nil
+	}
+
+	return project_update_lib.JobStatus{
+		Completed:             true,
+		PercentageComplete:    1,
+		EstimatedEndTimeInSec: 0,
+	}, nil
+}
+
+func (db *DB) updateNodes(projectRules map[string]*iam_v2.ProjectRules) {
+	nodes, err := db.getAllNodes()
+	if err != nil {
+		db.projectUpdateFail(err)
+		return
+	}
+
+	numberOfNodes := float32(len(nodes))
+	for index, node := range nodes {
+		if !db.ProjectUpdate.running {
+			db.projectUpdateComplete()
+			return
+		}
+		matchingProjectIDs := getMatchingProjectIDs(node, projectRules)
+
+		err = db.updateNodeProjectIDs(node, matchingProjectIDs)
+		if err != nil {
+			db.projectUpdateFail(err)
+			return
+		}
+
+		db.projectUpdateStatusUpdate(float32((index + 1)) / numberOfNodes)
+	}
+
+	db.projectUpdateComplete()
+}
```

Comment:
What this is doing is for each node in the node manager update their project IDs with the `projectRules map[string]*iam_v2.ProjectRules` rules. 

I am wondering if all this go code can be replaced with an SQL script.